### PR TITLE
Todolist tree view

### PR DESCRIPTION
**Pull request by adbharadwaj**
Current view (current backend doesnt return nested todo list):
screen shot 2018-06-18 at 5 50 50 pm

Once the backend starts returning nested todo list, the tree view will look like following:
screen shot 2018-06-20 at 10 11 21 am

Here is the response format required by the tree view template:
```
[{
        'id': 1,
        'description': 'THINGS NOT TO DO',
        'completed': false,
        'items': [
          {
            'id': 11,
            'description': 'node1.1',
            'completed': false,
            'items': [
              {
                'id': 111,
                'description': 'node1.1.1',
                'completed': false,
                'items': []
              }
            ]
          },
          {
            'id': 12,
            'description': 'node1.2',
            'completed': false,
            'items': []
          }
        ]
      },
      {
        'id': 2,
        'description': 'GETTING STARTED',
        'completed': false,
        'items': [
          {
            'id': 21,
            'description': 'node2.1',
            'completed': false,
            'items': []
          },
          {
            'id': 22,
            'description': 'node2.2',
            'completed': false,
            'items': []
          }
        ]
      },
      {
        'id': 3,
        'description': 'ADD/UPDATE CONTENT',
        'completed': false,
        'items': [
          {
            'id': 31,
            'description': 'node3.1',
            'completed': false,
            'items': []
          }
        ]
      }]
```

When the todo description is too long, the text-wrap looks like following:
screen shot 2018-06-20 at 1 25 22 pm

### CODE REVIEW

File: orchestra/static/dist/main.css [hunk 1]
```diff
@@ -2320,25 +2320,31 @@ a.logo {
     .todo-list .new-todo__description__datetime {
```
Comment (paopow) [inline orchestra/static/dist/main.css]: How does the line wrap when the description of a todo is long? Can you make a screenshot of that?

File: orchestra/static/orchestra/todos/todo-checklist.html [hunk 2]
```diff
@@ -1,12 +1,19 @@
-<div class="list-by-status" ng-if="(todos | filter: {skipped_datetime: showSkipped?'!!':'!'}).length > 0">
-  <!-- <h4>{{title}}</h4> -->
-  <div class="todo"
-       ng-repeat="todo in todos | filter: {skipped_datetime: showSkipped?'!!':'!'}">
+<script type="text/ng-template" id="nodes_renderer.html">
```
Comment (paopow) [inline orchestra/static/orchestra/todos/todo-checklist.html]: Can you record a loom video of dragging in action?

File: orchestra/static/orchestra/todos/todo-checklist.html [hunk 3]
```diff
@@ -1,12 +1,19 @@
...
+  <div ui-tree-handle class="tree-node tree-node-content">
     <label ng-class="{'text-danger': isInDanger(todo)}">
       <input type="checkbox"
              ng-hide="showSkipped"
              ng-model="todo.completed"
              ng-change="updateTodo(todo)">
+
```
Comment (paopow) [inline orchestra/static/orchestra/todos/todo-checklist.html]: What does the heirarchical skipped todo list look like? Can you either get a screenshot or show it a loom video?

File: orchestra/static/orchestra/todos/todo-checklist.html [hunk 4]
```diff
@@ -22,4 +29,17 @@
       </span>
     </label>
   </div>
+  <ol ui-tree-nodes="" ng-model="todo.items" ng-class="{hidden: collapsed}">
```
Comment (paopow) [inline orchestra/static/orchestra/todos/todo-checklist.html]: Empty prop?

File: orchestra/static/orchestra/todos/todo-checklist.html (hunk 2 above)
Comment (paopow) [inline orchestra/static/orchestra/todos/todo-checklist.html]: Is including the tree item template in a script tag a requirement for this library/angular? Is it possible to put them in their own `todo-item.html` file?

File: orchestra/static/orchestra/todos/todo-checklist.html [hunk 5]
```diff
@@ -22,4 +29,17 @@
...
+    <li ng-repeat="todo in todo.items | filter: {skipped_datetime: showSkipped?'!!':'!'}" ui-tree-node data-collapsed="true" ng-include="'nodes_renderer.html'"></li>
```
Comment (paopow) [inline orchestra/static/orchestra/todos/todo-checklist.html]: What does `showSkipped?'!!':'!'` mean?

File: orchestra/static/orchestra/todos/todo-checklist.html [hunk 6]
```diff
@@ -1,12 +1,19 @@
...
+
+      <i aria-hidden="true" ng-if="todo.items && todo.items.length > 0" data-nodrag
+        class="fa"
+        ng-class="{
+          'fa-caret-right': collapsed,
+          'fa-caret-down': !collapsed
+        }"></i>
+      <i class="fa fa-circle-o" aria-hidden="true" ng-if="(!todo.items || todo.items.length === 0) && showSkipped" data-nodrag></i>
+      <span ng-hide="!todo.items || todo.items.length < 1">[{{ ( todo.items | filter: { completed: true } ).length }}/{{ todo.items.length }}] </span>
       <span class="todo__role">{{steps[taskSlugs[todo.task]].name}}:&nbsp;</span>
```
Comment (paopow) [inline orchestra/static/orchestra/todos/todo-checklist.html]: Should we consider render the sub-todo in lighter font-weight? Will there be a limit on how deep the tree is?

File: orchestra/static/orchestra/todos/todo-checklist.html (hunk 2 above)
Comment (adbharadwaj) [reply orchestra/static/orchestra/todos/todo-checklist.html]: I have disabled dragging todos right now.

File: orchestra/static/orchestra/todos/todo-checklist.html (hunk 6 above)
Comment (adbharadwaj) [reply orchestra/static/orchestra/todos/todo-checklist.html]: Although `angular-ui-tree` allows us to enforce depth of a tree, I am not forcing any depth restrictions currently.

File: orchestra/static/orchestra/todos/todo-checklist.html (hunk 5 above)
Comment (adbharadwaj) [reply orchestra/static/orchestra/todos/todo-checklist.html]: `skipped_datetime: '!'` checks if the skipped_datetime is null

`skipped_datetime: '!!'` checks if the skipped_datetime is not null

`showSkipped?'!!':'!'` is choosing what to filter based on `showSkipped`

File: orchestra/static/orchestra/todos/todo-checklist.html (hunk 4 above)
Comment (adbharadwaj) [reply orchestra/static/orchestra/todos/todo-checklist.html]: removing it.

File: orchestra/static/orchestra/todos/todo-checklist.html (hunk 5 above)
Comment (marcua) [reply orchestra/static/orchestra/todos/todo-checklist.html]: if there's a less cryptic way to do this, i'm all for it :)

File: orchestra/static/orchestra/todos/todo-checklist.html (hunk 2 above)
Comment (marcua) [inline orchestra/static/orchestra/todos/todo-checklist.html]: do we need `<script type="text/ng-template" id="nodes_renderer.html">`?

File: orchestra/static/orchestra/todos/todo-checklist.scss [hunk 7]
```diff
@@ -1,35 +1,45 @@
...
-      label {
-        align-items: end;
-        display: flex;
-      }
-      &__role {
-        font-weight: 500;
-        padding-left: 5px;
+    label {
+      align-items: end;
+      // display: flex;
```
Comment (marcua) [inline orchestra/static/orchestra/todos/todo-checklist.scss]: clean up if not necessary

File: orchestra/static/orchestra/todos/todo-checklist.scss [hunk 8]
```diff
@@ -1,35 +1,45 @@
...
-        }
+      &-separator {
+        font-style: normal;
       }
-
     }
+
+  }
+  .angular-ui-tree-node {
+      padding-top: 5px !important;
```
Comment (marcua) [inline orchestra/static/orchestra/todos/todo-checklist.scss]: any way to avoid !important with more specific selectors?

File: orchestra/templates/orchestra/base.html [hunk 9]
```diff
@@ -29,7 +29,8 @@
     <link rel="stylesheet" type="text/css" href="{% static 'hijack/hijack-styles.css' %}" />
     <link rel="stylesheet" type="text/css" href="{% static 'dist/orchestra/common/css/orchestra.css' %}">
     <link rel="stylesheet" type="text/css" href="//cdnjs.cloudflare.com/ajax/libs/angular-ui-select/0.16.1/select.min.css">
-
+    <link rel="stylesheet" type="text/css" href="//cdnjs.cloudflare.com/ajax/libs/angular-ui-tree/2.22.6/angular-ui-tree.min.css">
```
Comment (marcua) [inline orchestra/templates/orchestra/base.html]: ensuring this can't be done w/ an ES6 import and a `yarn install`

File: orchestra/templates/orchestra/index.html [hunk 10]
```diff
@@ -29,6 +29,7 @@
 <script src="//ajax.googleapis.com/ajax/libs/angularjs/1.5.5/angular-sanitize.min.js"></script>
 <script src="//cdnjs.cloudflare.com/ajax/libs/angular-ui-bootstrap/1.3.1/ui-bootstrap-tpls.min.js"></script>
 <script src="//cdnjs.cloudflare.com/ajax/libs/angular-ui-select/0.16.1/select.min.js" type="text/javascript"></script>
+<script src="//cdnjs.cloudflare.com/ajax/libs/angular-ui-tree/2.22.6/angular-ui-tree.min.js" type="text/javascript"></script>
```
Comment (marcua) [inline orchestra/templates/orchestra/index.html]: ensuring this can't be done w/ an ES6 import and a `yarn install`

File: orchestra/static/dist/main.css [hunk 11]
```diff
@@ -2320,25 +2320,31 @@ a.logo {
...
       display: inline-block;
```
Comment (marcua) [inline orchestra/static/dist/main.css]: sanity-checks
- [x] due date still works
- [x] start date still works

File: orchestra/static/dist/main.css (hunk 1 above)
Comment (adbharadwaj) [reply orchestra/static/dist/main.css]: add the screenshot to the description.

Comment (paopow) [reply orchestra/static/dist/main.css]: Is it possible to make sure that the text don't wrap up under the box? Example: https://ps.w.org/cleverness-to-do-list/assets/screenshot-1.jpg?rev=1132235

Comment (adbharadwaj) [reply orchestra/static/dist/main.css]: Fixed it. I also updated the screenshots.

File: orchestra/static/dist/main.css [hunk 12]
```diff
@@ -2320,25 +2320,91 @@ a.logo {
...
-.existing-todos {
-  padding-top: 5px; }
-  .existing-todos .list-by-status {
-    padding-top: 20px; }
-    .existing-todos .list-by-status h4 {
-      padding-bottom: 10px; }
-    .existing-todos .list-by-status .todo input[type=checkbox] {
-      transform: scale(1.1); }
-    .existing-todos .list-by-status .todo label {
+.list-by-status {
```
Comment (marcua) [inline orchestra/static/dist/main.css]: Looking at the video, there's a weird amount of vertical padding above the skipped item list. Can you remove that?:)

Comment (adbharadwaj) [reply orchestra/static/dist/main.css]: I added that to keep both lists at the same level. But I am good with removing it too.

**Review by marcua — APPROVED** [review]

**Review by paopow — APPROVED** [review]
